app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,81 @@
+
+class TasksController < ApplicationController
+    def index
+        @tasks =Task.all
+    end
+
+    def show
+        task_id = params[:id]
+        @task = Task.find_by(id: task_id)
+        if @task.nil?
+            redirect_to root_path
+            return
+        end
+    end
+
+    def new
+        @task = Task.new
+    end
+
+    def create
+        @task = Task.new(
+            name: params[:task][:name],
+            description: params[:task][:description],
+            completed_at: params[:task][:completed_at]
+        )
```

Comment:
It wasn't a requirement for this project, but if you had more time on this project, I'd encourage you to use a strong params method!